src/V3SymTable.h

```diff
@@ -156,15 +156,47 @@ class VSymEnt final {
                      << (it == m_idNameMap.end() ? "NONE"
                                                  : "se" + cvtToHex(it->second)
                                                        + " n=" + cvtToHex(it->second->nodep())));
-        if (it != m_idNameMap.end()) return it->second;
+        if (it != m_idNameMap.end()) {
+            if (!classOrPackage) return it->second;
+            if (VN_IS(it->second->nodep(), Class) || VN_IS(it->second->nodep(), Package)) {
+                return it->second;
```

Comment:
V3SymTable is intended to be fairly generic, so it doesn't feel right to have this much node-specific code in it, I'd prefer the V3LinkDot resolve function get more compilcated or do some of this itself.

Also there's a lot of new branches in this code, are you sure that they are all covered by the test you added or older tests?